### PR TITLE
{Notification-hub} Migrate notification-hub to AAZ

### PR DESCRIPTION
**Pull request by Jing-song**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az notification-hub
AAZ model: https://github.com/Azure/aaz/pull/243

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Notification-hub

**Comment by kairu-ms** [comment]
image

**Comment by kairu-ms** [comment]
Please verify the old command needs confirm or not
image
@AllyW Do we have such check in breaking change tool?

**Comment by kairu-ms** [comment]
image

**Comment by kairu-ms** [comment]
image

**Comment by kairu-ms** [comment]
image
image

**Comment by kairu-ms** [comment]
image
image

**Comment by kairu-ms** [comment]
image
image
image
image
image

**Comment by Jing-song** [comment]
> Please verify the following option names are not changed  image
https://github.com/Azure/azure-rest-api-specs/blob/edd7863aeaad7379a7a8eef424ca11617d8c91eb/specification/notificationhubs/resource-manager/Microsoft.NotificationHubs/stable/2017-04-01/notificationhubs.json#L2158-L2161
`test-send` cannot generate these custom parameters, and cannot modify the request body

File: src/notification-hub/azext_notification_hub/commands.py [hunk 1]
```diff
@@ -7,59 +7,32 @@
...
+        self.command_table['notification-hub namespace authorization-rule regenerate-keys'] = NamespaceRuleRegenerateKeys(loader=self)
+
+    with self.command_group('notification-hub credential'):
+        from azext_notification_hub.custom import BaiduUpdate, ApnsUpdate, MpnsUpdate, AdmUpdate, WnsUpdate, GcmUpdate
+        self.command_table['notification-hub credential gcm update'] = GcmUpdate(loader=self)
+        self.command_table['notification-hub credential adm update'] = AdmUpdate(loader=self)
+        self.command_table['notification-hub credential apns update'] = ApnsUpdate(loader=self)
+        self.command_table['notification-hub credential wns update'] = WnsUpdate(loader=self)
+        self.command_table['notification-hub credential mpns update'] = MpnsUpdate(loader=self)
+        self.command_table['notification-hub credential baidu update'] = BaiduUpdate(loader=self)
```
Comment (kairu-ms) [inline src/notification-hub/azext_notification_hub/commands.py:38]: Why not use the aaz generated update command directly?

Comment (kairu-ms) [reply src/notification-hub/azext_notification_hub/commands.py:38]: image

File: src/notification-hub/azext_notification_hub/custom.py [hunk 2]
```diff
@@ -7,373 +7,163 @@
...
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.adm_credential._registered = False
+        args_schema.apns_credential._registered = False
+        args_schema.baidu_credential._registered = False
+        args_schema.gcm_credential._registered = False
+        args_schema.mpns_credential._registered = False
+        args_schema.wns_credential._registered = False
+        return args_schema
```
Comment (kairu-ms) [inline src/notification-hub/azext_notification_hub/custom.py:45]: You can keep them open

File: src/notification-hub/azext_notification_hub/custom.py [hunk 3]
```diff
@@ -7,373 +7,163 @@
...
-                           mpns_certificate,
-                           certificate_key):
-    import base64
-    body = {}
-    if mpns_certificate is not None:
-        with open(mpns_certificate, "rb") as f:
-            data_bytes = f.read()
-            cert_data = base64.b64encode(data_bytes).decode('utf-8')
-            body.setdefault('mpns_credential', {})['mpns_certificate'] = cert_data
-    body.setdefault('mpns_credential', {})['certificate_key'] = certificate_key
```
Comment (kairu-ms) [inline src/notification-hub/azext_notification_hub/custom.py:366]: It seems the import is base64 encoded file. Please change the argument of it.

File: src/notification-hub/azext_notification_hub/custom.py [hunk 4]
```diff
@@ -7,373 +7,163 @@
...
-                           app_name=None,
-                           app_id=None,
-                           token=None):
-    import base64
-    body = {}
-    if apns_certificate is not None:
-        with open(apns_certificate, "rb") as f:
-            data_bytes = f.read()
-            cert_data = base64.b64encode(data_bytes).decode('utf-8')
-            body.setdefault('apns_credential', {})['apns_certificate'] = cert_data
```
Comment (kairu-ms) [inline src/notification-hub/azext_notification_hub/custom.py:325]: Same here

File: src/notification-hub/azext_notification_hub/commands.py [hunk 5]
```diff
@@ -7,59 +7,32 @@
...
-        g.custom_command('regenerate-keys', 'regenerate_keys_notificationhubs_hub')
-
-    with self.command_group('notification-hub credential', notificationhubs_notification_hubs, client_factory=cf_notification_hubs) as g:
-        g.custom_command('list', 'get_pns_credentials_notificationhubs_hub')
-        g.custom_command('gcm update', 'update_gcm_credential')
-        g.custom_command('apns update', 'update_apns_credential')
-        g.custom_command('wns update', 'update_wns_credential')
-        g.custom_command('mpns update', 'update_mpns_credential')
-        g.custom_command('adm update', 'update_adm_credential')
-        g.custom_command('baidu update', 'update_baidu_credential')
```
Comment (kairu-ms) [inline src/notification-hub/azext_notification_hub/commands.py:65]: According the the definition, there's not create command, but only update command.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**